src/components/sbb-navigation/sbb-navigation.scss

```diff
@@ -75,27 +55,22 @@
   }
 }
 
-:host(:not(.sbb-navigation-section--closed)) {
-  inset: 0;
+:host([data-state='opened']) {
+  --sbb-navigation-animation-easing: ease-out;
 }
 
-:host(.sbb-navigation--opened),
-:host(.sbb-navigation--opening) {
-  @include sbb.mq($from: large) {
-    --sbb-navigation-backdrop-color: var(--sbb-color-black-alpha-70);
-
-    &::before {
-      visibility: visible;
-      pointer-events: all;
-    }
-  }
+:host(:is([data-state='opened'], [data-state='opening'])) {
+  --sbb-navigation-backdrop-visibility: visible;
+  --sbb-navigation-backdrop-pointer-events: all;
+  --sbb-navigation-backdrop-color: var(--sbb-color-black-alpha-70);
 }
 
-:host(.sbb-navigation--opened) {
-  --sbb-navigation-animation-easing: ease-out;
+:host(:not([data-state='closed'])) {
+  --sbb-navigation-inset: 0;
+  --sbb-navgation-transform: translateX(0);
 }
 
-:host(.sbb-navigation--has-navigation-section) {
+:host([data-has-navigation-section]) {
```

Comment:
I think the way you implemented it on tsx file, it should also handle the false case here? (:not([data-has-navigation-section="false"])

Reply:
This shouldn't actually happen, as this attribute is added/removed from within the component and should not be exposed to the consumer. However, we can add the rule anyway if you prefer.

Reply:
I'm surprised that it is removed like it's implemented :D

Reply:
but yes when it's working internally, no other case should be checked